From 7c67b4cf33d1c209e1796cd78fc943f2c98379dd Mon Sep 17 00:00:00 2001
From: Sandy Huang <hjc@rock-chips.com>
Date: Thu, 2 Jul 2020 10:46:38 +0800
Subject: [PATCH] video: rockchip: lcdc: fix scale register config error

rk312x SCALER_CTRL[0xa0] offset is same with rk3036 WIN1_MST[0xa0], this
bug will lead to SCALER_CTRL register config error and lead to LCD
display splash screen.

Signed-off-by: Sandy Huang <hjc@rock-chips.com>
Change-Id: I3e07a64ede27a0976f72c0895e29ec6a3ddf4963
---
 drivers/video/rockchip/lcdc/rk312x_lcdc.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/lcdc/rk312x_lcdc.c b/drivers/video/rockchip/lcdc/rk312x_lcdc.c
index 8bd2111..a9b377c 100755
--- a/drivers/video/rockchip/lcdc/rk312x_lcdc.c
+++ b/drivers/video/rockchip/lcdc/rk312x_lcdc.c
@@ -542,7 +542,10 @@ static void lcdc_layer_update_regs(struct lcdc_device *lcdc_dev,
 		} else if (win->id == 1) {
 			lcdc_msk_reg(lcdc_dev, SYS_CTRL, m_WIN1_EN,
 				     v_WIN1_EN(0));
-			 lcdc_writel(lcdc_dev, WIN1_MST, win->area[0].y_addr);
+			if (lcdc_dev->soc_type == VOP_RK3036)
+				lcdc_writel(lcdc_dev, WIN1_MST, win->area[0].y_addr);
+			else
+				lcdc_writel(lcdc_dev, WIN1_MST_RK312X, win->area[0].y_addr);
 		} else if (win->id == 2) {
 			lcdc_msk_reg(lcdc_dev,
 			             SYS_CTRL, m_HWC_EN | m_HWC_LODAD_EN,
-- 
2.7.4

